tests/test_config.py

```diff
@@ -1007,23 +1016,72 @@ def test_no_toml_installed_pyproject_no_coverage(self) -> None:
             assert not cov.config.branch
             assert cov.config.data_file == ".coverage"
 
-    def test_exceptions_from_missing_toml_things(self) -> None:
+    @pytest.mark.parametrize("filename", ["pyproject.toml", ".coveragerc.toml"])
+    def test_exceptions_from_missing_toml_things(self, filename: str) -> None:
         self.make_file(
-            "pyproject.toml",
+            filename,
             """\
             [tool.coverage.run]
             branch = true
             """,
         )
         config = TomlConfigParser(False)
-        config.read("pyproject.toml")
+        config.read(filename)
         with pytest.raises(ConfigError, match="No section: 'xyzzy'"):
             config.options("xyzzy")
         with pytest.raises(ConfigError, match="No section: 'xyzzy'"):
             config.get("xyzzy", "foo")
         with pytest.raises(ConfigError, match="No option 'foo' in section: 'tool.coverage.run'"):
             config.get("run", "foo")
 
+    def test_coveragerc_toml_priority(self) -> None:
+        """Test that .coveragerc.toml has priority over pyproject.toml."""
+        self.make_file(
+            ".coveragerc.toml",
+            """\
+            [tool.coverage.run]
+            timid = true
+            data_file = ".toml-data.dat"
+            branch = true
+            """,
+        )
+
+        self.make_file(
+            "pyproject.toml",
+            """\
+            [tool.coverage.run]
+            timid = false
+            data_file = "pyproject-data.dat"
+            branch = false
+            """,
+        )
+        cov = coverage.Coverage()
+
+        assert cov.config.timid is True
+        assert cov.config.data_file == ".toml-data.dat"
+        assert cov.config.branch is True
+
+    def test_coveragerc_toml_unprefixed_syntax(self) -> None:
```

Comment:
It'd be a good idea to also add a test making sure that unprefixed sections are ignored if present in `pyproject.toml`.